tests/api_tests/comments/test_views.py

```diff
@@ -0,0 +1,889 @@
+        assert_equal(res.status_code, 403)
+
+    def test_return_private_node_comment_replies_logged_in_contributor(self):
+        res = self.app.get(self.private_url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        comment_json = res.json['data']
+        comment_ids = [comment['id'] for comment in comment_json]
+        assert_equal(len(comment_json), 1)
+        assert_in(self.comment_reply._id, comment_ids)
+
+    def test_return_public_node_comment_replies_logged_out_user(self):
+        res = self.app.get(self.public_url)
+        assert_equal(res.status_code, 200)
+        comment_json = res.json['data']
+        comment_ids = [comment['id'] for comment in comment_json]
+        assert_equal(len(comment_json), 1)
+        assert_in(self.public_comment_reply._id, comment_ids)
+
+    def test_return_public_node_comment_replies_logged_in_non_contributor(self):
+        res = self.app.get(self.public_url, auth=self.non_contributor)
+        assert_equal(res.status_code, 200)
+        comment_json = res.json['data']
+        comment_ids = [comment['id'] for comment in comment_json]
+        assert_equal(len(comment_json), 1)
+        assert_in(self.public_comment_reply._id, comment_ids)
+
+    def test_return_public_node_comment_replies_logged_in_contributor(self):
+        res = self.app.get(self.public_url, auth=self.user)
+        assert_equal(res.status_code, 200)
+        comment_json = res.json['data']
+        comment_ids = [comment['id'] for comment in comment_json]
+        assert_equal(len(comment_json), 1)
+        assert_in(self.public_comment_reply._id, comment_ids)
+
+    def test_return_registration_comment_replies_logged_in_contributor(self):
+        res = self.app.get(self.registration_url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        comment_json = res.json['data']
+        comment_ids = [comment['id'] for comment in comment_json]
+        assert_equal(len(comment_json), 1)
+        assert_in(self.registration_comment_reply._id, comment_ids)
+
+    def test_return_both_deleted_and_undeleted_comment_replies(self):
+        deleted_comment_reply = CommentFactory(project=self.private_project, target=self.comment, user=self.user, is_deleted=True)
+        res = self.app.get(self.private_url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        comment_json = res.json['data']
+        comment_ids = [comment['id'] for comment in comment_json]
+        assert_in(self.comment_reply._id, comment_ids)
+        assert_in(deleted_comment_reply._id, comment_ids)
+
+
+class TestCommentRepliesCreate(ApiTestCase):
+    def setUp(self):
+        super(TestCommentRepliesCreate, self).setUp()
+        self.user = AuthUserFactory()
+        self.read_only_contributor = AuthUserFactory()
+        self.non_contributor = AuthUserFactory()
+
+        self.private_project = ProjectFactory(is_public=False, creator=self.user)
```

Comment:
This is shorthand for `ProjectFactory.create()`, which generates and saves the instance immediately. Consider using `ProjectFactory.build()` since it is being explicitly saved again below- avoiding redundant database hits might help to speed up tests. 

https://factoryboy.readthedocs.org/en/latest/#using-factories